sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -413,7 +414,10 @@ private[hive] class HiveClientImpl(
         unsupportedFeatures += "partitioned view"
       }
 
-      val properties = Option(h.getParameters).map(_.asScala.toMap).orNull
+      val properties = Option(h.getParameters).map(_.asScala.toMap).getOrElse(Map())
+
+      val provider = properties.get(HiveExternalCatalog.DATASOURCE_PROVIDER)
+        .orElse(Some(DDLUtils.HIVE_PROVIDER))
```

Comment:
> Maybe this is redundant.

This was definitely not redundant in my testing. The metadata loaded from the metastore in `HiveExternalCatalog.alterTableSchema` was definitely not set when I debugged this. In fact the test I wrote fails if I remove this code (or comment the line that sets "provider" a few lines below).

Perhaps some other part of the code sets it in a different code path, but this would make that part of the code redundant, not the other way around.